<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1.0">
<!-- Inform modern browsers that this page supports both dark and light color schemes,
  and the page author prefers light. --><meta name="color-scheme" content="dark light">
<script>
  // If `prefers-color-scheme` is not supported, fall back to light mode.
  // i.e. In this case, inject the `light` CSS before the others, with
  // no media filter so that it will be downloaded with highest priority.
  if (window.matchMedia("(prefers-color-scheme: dark)").media === "not all") {
    document.documentElement.style.display = "none";
    document.head.insertAdjacentHTML(
      "beforeend",
      "<link id=\"css\" rel=\"stylesheet\" href=\"https://bootswatch.com/3/flatly/bootstrap.css\" onload=\"document.documentElement.style.display = ''\">"
    );
  }
</script><title>Get started with Visium Brain data • spots</title>
<!-- favicons --><link rel="icon" type="image/png" sizes="16x16" href="../favicon-16x16.png">
<link rel="icon" type="image/png" sizes="32x32" href="../favicon-32x32.png">
<link rel="apple-touch-icon" type="image/png" sizes="180x180" href="../apple-touch-icon.png">
<link rel="apple-touch-icon" type="image/png" sizes="120x120" href="../apple-touch-icon-120x120.png">
<link rel="apple-touch-icon" type="image/png" sizes="76x76" href="../apple-touch-icon-76x76.png">
<link rel="apple-touch-icon" type="image/png" sizes="60x60" href="../apple-touch-icon-60x60.png">
<!-- jquery --><script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.4.1/jquery.min.js" integrity="sha256-CSXorXvZcTkaix6Yvo6HppcZGetbYMGWSFlBw8HfCJo=" crossorigin="anonymous"></script><!-- Bootstrap --><script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/js/bootstrap.min.js" integrity="sha256-nuL8/2cJ5NDSSwnKD8VqreErSWHtnEP9E7AySL+1ev4=" crossorigin="anonymous"></script><!-- bootstrap-toc --><link rel="stylesheet" href="../bootstrap-toc.css">
<script src="../bootstrap-toc.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous">
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous">
<!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- pkgdown --><link href="../pkgdown.css" rel="stylesheet">
<script src="../pkgdown.js"></script><meta property="og:title" content="Get started with Visium Brain data">
<meta property="og:description" content="spots">
<meta property="og:image" content="https://stevexniu.github.io/rspca/logo.png">
<!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]--><!-- Flatly Theme - Light  --><link id="css-light" rel="stylesheet" href="https://bootswatch.com/3/flatly/bootstrap.css" media="(prefers-color-scheme: light), (prefers-color-scheme: no-preference)">
<!-- Darkly Theme - Dark --><link id="css-dark" rel="stylesheet" href="https://bootswatch.com/3/darkly/bootstrap.css" media="(prefers-color-scheme: dark)">
<!-- preferably CSS --><link rel="stylesheet" href="../preferably.css">
<link id="css-code-light" rel="stylesheet" href="../code-color-scheme-light.css" media="(prefers-color-scheme: light), (prefers-color-scheme: no-preference)">
<link id="css-code-dark" rel="stylesheet" href="../code-color-scheme-dark.css" media="(prefers-color-scheme: dark)">
<script src="../darkswitch.js"></script>
</head>
<body data-spy="scroll" data-target="#toc">
    

    <div class="container template-article">
      <header><div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../index.html">spots</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="">0.1.0</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" data-bs-toggle="dropdown" aria-expanded="false">
    Vignettes
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
<li>
      <a href="../articles/get_started.html">Get started with Visium Brain data</a>
    </li>
  </ul>
</li>
<li>
  <a href="../reference/index.html">Reference</a>
</li>
<li>
  <a href="../news/index.html">News</a>
</li>
      </ul>
<ul class="nav navbar-nav navbar-right">
<li>
          <a href="#" id="css-toggle-btn">
            <span class="fas fa-adjust fa-lg"></span>
          </a>
        </li>
        
        <li>
  <a href="https://github.com/stevexniu/rspca/" class="external-link">
    <span class="fab fa-github fa-lg"></span>
     
  </a>
</li>
        
        


      </ul>
</div>
<!--/.nav-collapse -->
  </div>
<!--/.container -->
</div>
<!--/.navbar -->

      

      </header><div class="row">
  <div class="col-md-9 contents">
    <div class="page-header toc-ignore">
      <h1 data-toc-skip>Get started with Visium Brain data</h1>
            
            <h4 data-toc-skip class="date">Last Run: April 29, 2022</h4>
      
      <small class="dont-index">Source: <a href="https://github.com/stevexniu/rspca/blob/HEAD/vignettes/get_started.Rmd" class="external-link"><code>vignettes/get_started.Rmd</code></a></small>
      <div class="hidden name"><code>get_started.Rmd</code></div>

    </div>

    
    
<div class="section level3">
<h3 id="load-visium-brain-data">Load Visium Brain Data<a class="anchor" aria-label="anchor" href="#load-visium-brain-data"></a>
</h3>
<p>For this tutorial, we will be analyzing the <a href="https://support.10xgenomics.com/spatial-gene-expression/datasets" class="external-link">Visium Sagittal-Posterior Mouse Brain</a> data produced by 10x Genomics.</p>
<p>A fraction of this data containing 3353 spatial barcodes (spots) and 2000 highly variable genes can be loaded using the <code>LoadData</code> function. We will be using <a href="https://satijalab.org/seurat/articles/spatial_vignette.html" class="external-link">Seurat</a> for this analysis.</p>
<div class="sourceCode"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co"># install Seurat v4.0.0</span>
<span class="kw">if</span> <span class="op">(</span><span class="op">!</span><span class="fu"><a href="https://rdrr.io/r/base/ns-load.html" class="external-link">requireNamespace</a></span><span class="op">(</span><span class="st">"Seurat"</span>, quietly <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span> <span class="op">|</span> <span class="fu">utils</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/r/utils/packageDescription.html" class="external-link">packageVersion</a></span><span class="op">(</span><span class="st">"Seurat"</span><span class="op">)</span> <span class="op">&lt;</span> <span class="st">"4.0.0"</span><span class="op">)</span>
  <span class="fu">remotes</span><span class="fu">::</span><span class="fu"><a href="https://remotes.r-lib.org/reference/install_version.html" class="external-link">install_version</a></span><span class="op">(</span><span class="st">"Seurat"</span>, version <span class="op">=</span> <span class="st">"4.0.0"</span><span class="op">)</span>

<span class="co"># load data</span>
<span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va">rspca</span><span class="op">)</span>
<span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://satijalab.org/seurat" class="external-link">Seurat</a></span><span class="op">)</span>
<span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://ggplot2.tidyverse.org" class="external-link">ggplot2</a></span><span class="op">)</span>
<span class="va">posterior1</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/LoadData.html">LoadData</a></span><span class="op">(</span><span class="st">"Visium.Brain"</span><span class="op">)</span>
<span class="fu"><a href="https://satijalab.org/seurat/reference/SpatialPlot.html" class="external-link">SpatialDimPlot</a></span><span class="op">(</span><span class="va">posterior1</span><span class="op">)</span></code></pre></div>
<p><img src="get_started_files/figure-html/load-1.png" width="768"></p>
</div>
<div class="section level3">
<h3 id="spatial-component-analysis-sca">Spatial Component Analysis (SCA)<a class="anchor" aria-label="anchor" href="#spatial-component-analysis-sca"></a>
</h3>
<p>To run Spatial Component Analysis (SCA), we will need to define a weight matrix between each pair of datapoints (spots).</p>
<p>To achieve this, we first calculate the distance between each spots using hexagonal nearest neighbor distance (<code>VisiumHnn</code> and <code>HnnNeighbor</code> functions) as it resembles the way Visium slides were generated.</p>
Typically, datapoints not located on the edges should have 6 1st-degree, 12 2nd-degree, 18 3rd-degree neighbors and so on. These specific parameters can be controlled using <code>HnnNeighbor</code> function.
<center>
<img src="hnn_image.png" alt="hnn" width="400">
</center>
<p>To turn this distance matrix to a weight matrix, we call <code>HnnWeight</code> function to apply a Gaussian filter. I'm using parameter <code>dist.k = 2</code> to consider only the 2nd-degree neighbors and <code>sigma = 2</code> for filter width.</p>
<div class="sourceCode"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co"># HNN distance </span>
<span class="va">posterior1.hnn.dist</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/VisiumHnn.html">VisiumHnn</a></span><span class="op">(</span><span class="fu"><a href="https://satijalab.org/seurat/reference/Cells.html" class="external-link">Cells</a></span><span class="op">(</span><span class="va">posterior1</span><span class="op">)</span><span class="op">)</span>

<span class="co"># HNN weights</span>
<span class="va">posterior1.data.hnn</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/HnnNeighbor.html">HnnNeighbor</a></span><span class="op">(</span><span class="va">posterior1.hnn.dist</span>, k <span class="op">=</span> <span class="fl">19</span>, include.self <span class="op">=</span> <span class="cn">FALSE</span><span class="op">)</span>
<span class="va">posterior1.data.weight</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/HnnWeight.html">HnnWeight</a></span><span class="op">(</span><span class="va">posterior1.data.hnn</span><span class="op">$</span><span class="va">dist.mat</span>, dist.k <span class="op">=</span> <span class="fl">2</span>, sigma <span class="op">=</span> <span class="fl">2</span><span class="op">)</span></code></pre></div>
<p>After this, we are ready to run SCA using the 2000 spatial transcriptomic measurements and the weight matrix we just calculated. Here I set the <code>n.eigen = 30</code> to return only the first 30 Spatial Components (SCs), and plot four different visualizations:</p>
<ol style="list-style-type: decimal">
<li>Dimensional reduction plot of first two SCs. (top left)</li>
<li>Elbow plot showing the eigenvalues or variance explained. (top right)</li>
<li>Visualization of the first SC levels in space. (bottom left)</li>
<li>Heatmap showing the features with the largest SC loadings. (bottom right)</li>
</ol>
<div class="sourceCode"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co"># Run SCA</span>
<span class="va">posterior1</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://satijalab.org/seurat/reference/ScaleData.html" class="external-link">ScaleData</a></span><span class="op">(</span><span class="va">posterior1</span><span class="op">)</span>
<span class="va">posterior1.sca</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/SCA.html">SCA</a></span><span class="op">(</span>X <span class="op">=</span> <span class="fu">Matrix</span><span class="fu">::</span><span class="fu">t</span><span class="op">(</span><span class="va">posterior1</span><span class="op">@</span><span class="va">assays</span><span class="op">$</span><span class="va">Spatial</span><span class="op">@</span><span class="va">data</span><span class="op">)</span>, 
                      W <span class="op">=</span> <span class="va">posterior1.data.weight</span>, 
                      scaled.data <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/t.html" class="external-link">t</a></span><span class="op">(</span><span class="va">posterior1</span><span class="op">@</span><span class="va">assays</span><span class="op">$</span><span class="va">Spatial</span><span class="op">@</span><span class="va">scale.data</span><span class="op">)</span>,
                      n.eigen <span class="op">=</span> <span class="fl">30</span><span class="op">)</span>

<span class="co"># Store in the Seurat object</span>
<span class="va">posterior1</span><span class="op">@</span><span class="va">reductions</span><span class="op">[[</span><span class="st">"sca"</span><span class="op">]</span><span class="op">]</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://satijalab.org/seurat/reference/CreateDimReducObject.html" class="external-link">CreateDimReducObject</a></span><span class="op">(</span>embeddings <span class="op">=</span> <span class="va">posterior1.sca</span><span class="op">$</span><span class="va">X</span>, 
                                                       loadings <span class="op">=</span> <span class="va">posterior1.sca</span><span class="op">$</span><span class="va">rotation</span>, 
                                                       stdev <span class="op">=</span> <span class="va">posterior1.sca</span><span class="op">$</span><span class="va">eigenvalues</span>, 
                                                       key <span class="op">=</span> <span class="st">"SC_"</span>, assay <span class="op">=</span> <span class="st">"Spatial"</span><span class="op">)</span>

<span class="co"># Visualization</span>
<span class="va">p1</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://satijalab.org/seurat/reference/DimPlot.html" class="external-link">DimPlot</a></span><span class="op">(</span><span class="va">posterior1</span>, reduction <span class="op">=</span> <span class="st">"sca"</span>, group.by <span class="op">=</span> <span class="st">"region"</span><span class="op">)</span>
<span class="va">p2</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://satijalab.org/seurat/reference/ElbowPlot.html" class="external-link">ElbowPlot</a></span><span class="op">(</span><span class="va">posterior1</span>, reduction <span class="op">=</span> <span class="st">"sca"</span>, ndims <span class="op">=</span> <span class="fl">30</span><span class="op">)</span>
<span class="va">p3</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://satijalab.org/seurat/reference/SpatialPlot.html" class="external-link">SpatialFeaturePlot</a></span><span class="op">(</span><span class="va">posterior1</span>, features <span class="op">=</span> <span class="st">'SC_1'</span><span class="op">)</span>
<span class="va">p4</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://satijalab.org/seurat/reference/DimHeatmap.html" class="external-link">DimHeatmap</a></span><span class="op">(</span><span class="va">posterior1</span>, reduction <span class="op">=</span> <span class="st">"sca"</span>, fast <span class="op">=</span> <span class="cn">FALSE</span><span class="op">)</span>
<span class="op">(</span><span class="va">p1</span><span class="op">+</span><span class="va">p2</span><span class="op">)</span><span class="op">/</span><span class="op">(</span><span class="va">p3</span><span class="op">+</span><span class="va">p4</span><span class="op">)</span></code></pre></div>
<p><img src="get_started_files/figure-html/sca-1.png" width="1152"></p>
</div>
<div class="section level3">
<h3 id="downstream-analysis">Downstream Analysis<a class="anchor" aria-label="anchor" href="#downstream-analysis"></a>
</h3>
<p>Once finished, we can use the Spatial Components (SCs) for downstream analysis such as clustering. Here we utilized Seurat's <code>FindClusters</code> function to identify clusters of spots using the first 20 SCs we just calculated.</p>
<div class="sourceCode"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">posterior1</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://satijalab.org/seurat/reference/FindNeighbors.html" class="external-link">FindNeighbors</a></span><span class="op">(</span><span class="va">posterior1</span>, reduction <span class="op">=</span> <span class="st">"sca"</span>, dims <span class="op">=</span> <span class="fl">1</span><span class="op">:</span><span class="fl">20</span>,graph.name <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"sca.nn"</span>, <span class="st">"sca.snn"</span><span class="op">)</span><span class="op">)</span>
<span class="va">posterior1</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://satijalab.org/seurat/reference/FindClusters.html" class="external-link">FindClusters</a></span><span class="op">(</span><span class="va">posterior1</span>, resolution <span class="op">=</span> <span class="fl">1</span>, verbose <span class="op">=</span> <span class="cn">FALSE</span>, graph.name <span class="op">=</span> <span class="st">"sca.snn"</span><span class="op">)</span>
<span class="fu"><a href="https://satijalab.org/seurat/reference/SpatialPlot.html" class="external-link">SpatialDimPlot</a></span><span class="op">(</span><span class="va">posterior1</span>, label <span class="op">=</span> <span class="cn">TRUE</span>, label.size <span class="op">=</span> <span class="fl">7</span>, repel <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span></code></pre></div>
<p><img src="get_started_files/figure-html/clustering-1.png" width="768"></p>
</div>
<div class="section level3">
<h3 id="comparison-to-pca">Comparison to PCA<a class="anchor" aria-label="anchor" href="#comparison-to-pca"></a>
</h3>
<p>We can compare the clustering result based on SCA to PCA. Here I used the same clustering parameters but with the first 20 PCs instead of SCs.</p>
<div class="sourceCode"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">posterior1</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://satijalab.org/seurat/reference/RunPCA.html" class="external-link">RunPCA</a></span><span class="op">(</span><span class="va">posterior1</span>, npcs <span class="op">=</span> <span class="fl">30</span>, verbose <span class="op">=</span> <span class="cn">FALSE</span><span class="op">)</span>
<span class="va">posterior1</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://satijalab.org/seurat/reference/FindNeighbors.html" class="external-link">FindNeighbors</a></span><span class="op">(</span><span class="va">posterior1</span>, reduction <span class="op">=</span> <span class="st">"pca"</span>, dims <span class="op">=</span> <span class="fl">1</span><span class="op">:</span><span class="fl">20</span>, graph.name <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"pca.nn"</span>, <span class="st">"pca.snn"</span><span class="op">)</span><span class="op">)</span>
<span class="va">posterior1</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://satijalab.org/seurat/reference/FindClusters.html" class="external-link">FindClusters</a></span><span class="op">(</span><span class="va">posterior1</span>, resolution <span class="op">=</span> <span class="fl">1</span>, verbose <span class="op">=</span> <span class="cn">FALSE</span>, graph.name <span class="op">=</span> <span class="st">"pca.snn"</span><span class="op">)</span>
<span class="va">p1</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://satijalab.org/seurat/reference/SpatialPlot.html" class="external-link">SpatialDimPlot</a></span><span class="op">(</span><span class="va">posterior1</span>, label <span class="op">=</span> <span class="cn">TRUE</span>, label.size <span class="op">=</span> <span class="fl">5</span>, group.by <span class="op">=</span> <span class="st">"sca.snn_res.1"</span>, repel <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span> 
<span class="va">p2</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://satijalab.org/seurat/reference/SpatialPlot.html" class="external-link">SpatialDimPlot</a></span><span class="op">(</span><span class="va">posterior1</span>, label <span class="op">=</span> <span class="cn">TRUE</span>, label.size <span class="op">=</span> <span class="fl">5</span>, group.by <span class="op">=</span> <span class="st">"pca.snn_res.1"</span>, repel <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span> 
<span class="va">p1</span><span class="op">+</span><span class="va">p2</span></code></pre></div>
<p><img src="get_started_files/figure-html/pca-1.png" width="960"></p>
<p>There are 19 clusters (0-18) and 17 clusters (0-16) that were identified by SCA and PCA respectively.</p>
<p>We can visualize the differences between the two clustering results using Chi-square test and interpret the result with Chi-square residuals.</p>
<div class="sourceCode"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co"># Chi-square test</span>
<span class="va">tab</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/stats/xtabs.html" class="external-link">xtabs</a></span><span class="op">(</span>data <span class="op">=</span> <span class="va">posterior1</span><span class="op">@</span><span class="va">meta.data</span><span class="op">[</span>,<span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"sca.snn_res.1"</span>,<span class="st">"pca.snn_res.1"</span><span class="op">)</span><span class="op">]</span><span class="op">)</span>
<span class="va">chi.result</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/stats/chisq.test.html" class="external-link">chisq.test</a></span><span class="op">(</span><span class="va">tab</span><span class="op">)</span>
<span class="va">chi.resid</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/class.html" class="external-link">unclass</a></span><span class="op">(</span><span class="va">chi.result</span><span class="op">$</span><span class="va">residuals</span><span class="op">)</span>

<span class="co"># Data for visualization</span>
<span class="va">data.plot</span> <span class="op">&lt;-</span> <span class="fu">Seurat</span><span class="fu">:::</span><span class="fu">Melt</span><span class="op">(</span><span class="va">chi.resid</span><span class="op">)</span>
<span class="fu"><a href="https://rdrr.io/r/base/colnames.html" class="external-link">colnames</a></span><span class="op">(</span><span class="va">data.plot</span><span class="op">)</span><span class="op">[</span><span class="fl">1</span><span class="op">:</span><span class="fl">3</span><span class="op">]</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"SCA clustering"</span>, <span class="st">"PCA clustering"</span>, <span class="st">"Chi-square Residual"</span><span class="op">)</span>

<span class="co"># Reorder the columns</span>
<span class="va">data.plot</span><span class="op">[</span>,<span class="fl">1</span><span class="op">]</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/factor.html" class="external-link">factor</a></span><span class="op">(</span><span class="va">data.plot</span><span class="op">[</span>,<span class="fl">1</span><span class="op">]</span>, 
                        levels <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/order.html" class="external-link">order</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/apply.html" class="external-link">apply</a></span><span class="op">(</span><span class="va">chi.resid</span>, <span class="fl">1</span>, <span class="va">which.max</span><span class="op">)</span><span class="op">)</span><span class="op">[</span><span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="fl">1</span><span class="op">:</span><span class="fl">10</span>,<span class="fl">12</span><span class="op">:</span><span class="fl">11</span>,<span class="fl">13</span><span class="op">:</span><span class="fl">19</span><span class="op">)</span><span class="op">]</span><span class="op">-</span><span class="fl">1</span>, 
                        ordered <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span>
<span class="va">data.plot</span><span class="op">[</span>,<span class="fl">2</span><span class="op">]</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/factor.html" class="external-link">factor</a></span><span class="op">(</span><span class="va">data.plot</span><span class="op">[</span>,<span class="fl">2</span><span class="op">]</span>, levels <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/unique.html" class="external-link">unique</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/numeric.html" class="external-link">as.numeric</a></span><span class="op">(</span><span class="va">data.plot</span><span class="op">[</span>,<span class="fl">2</span><span class="op">]</span><span class="op">)</span><span class="op">)</span>, ordered <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span>

<span class="co"># Heatmap</span>
<span class="fu"><a href="https://ggplot2.tidyverse.org/reference/ggplot.html" class="external-link">ggplot</a></span><span class="op">(</span>data <span class="op">=</span> <span class="va">data.plot</span>, <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/aes.html" class="external-link">aes</a></span><span class="op">(</span>x<span class="op">=</span><span class="va">`PCA clustering`</span>, y<span class="op">=</span><span class="va">`SCA clustering`</span>, fill<span class="op">=</span><span class="va">`Chi-square Residual`</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span> 
  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/geom_tile.html" class="external-link">geom_tile</a></span><span class="op">(</span><span class="op">)</span> <span class="op">+</span> <span class="fu">cowplot</span><span class="fu">::</span><span class="fu"><a href="https://wilkelab.org/cowplot/reference/theme_cowplot.html" class="external-link">theme_cowplot</a></span><span class="op">(</span>font_size <span class="op">=</span> <span class="fl">20</span><span class="op">)</span> <span class="op">+</span> <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/labs.html" class="external-link">labs</a></span><span class="op">(</span>fill<span class="op">=</span><span class="st">"Chi-square\nResidual"</span><span class="op">)</span></code></pre></div>
<p><img src="get_started_files/figure-html/compare-1.png" width="768"></p>
</div>
<div class="section level3">
<h3 id="conclusion">Conclusion<a class="anchor" aria-label="anchor" href="#conclusion"></a>
</h3>
<p>For this data, the clustering results based on SCA and PCA in general agree with each other. However, as we noted PCA cluster 4, 9 and 10 were split by the SCA each into two distinct groups.</p>
<p>Here, we will go through one of them in detail to check whether SCA made a false discovery or PCA underestimated certain distinct clusters. <details><summary><strong>PCA Cluster 4 vs SCA Cluster 10 and 14</strong>&lt;&gt;</summary></details></p>
<div class="sourceCode"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">cluster4</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/subset.html" class="external-link">subset</a></span><span class="op">(</span><span class="va">posterior1</span>, <span class="va">pca.snn_res.1</span> <span class="op">==</span> <span class="fl">4</span> <span class="op">&amp;</span> <span class="va">sca.snn_res.1</span> <span class="op"><a href="https://rdrr.io/r/base/match.html" class="external-link">%in%</a></span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="fl">10</span>, <span class="fl">14</span><span class="op">)</span><span class="op">)</span>
<span class="va">cluster4</span><span class="op">$</span><span class="va">sca.snn_res.1</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/factor.html" class="external-link">factor</a></span><span class="op">(</span><span class="va">cluster4</span><span class="op">$</span><span class="va">sca.snn_res.1</span><span class="op">)</span>
<span class="va">cluster4</span><span class="op">$</span><span class="va">pca.snn_res.1</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/factor.html" class="external-link">factor</a></span><span class="op">(</span><span class="va">cluster4</span><span class="op">$</span><span class="va">pca.snn_res.1</span><span class="op">)</span>
<span class="va">p1</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://satijalab.org/seurat/reference/SpatialPlot.html" class="external-link">SpatialDimPlot</a></span><span class="op">(</span><span class="va">cluster4</span>, group.by <span class="op">=</span> <span class="st">"pca.snn_res.1"</span>, pt.size.factor <span class="op">=</span> <span class="fl">3</span>, 
                     label <span class="op">=</span> <span class="cn">TRUE</span>, label.size <span class="op">=</span> <span class="fl">5</span><span class="op">)</span> <span class="op">+</span> 
  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/labs.html" class="external-link">ggtitle</a></span><span class="op">(</span><span class="st">"PCA Clustering"</span><span class="op">)</span> <span class="op">+</span> <span class="fu"><a href="https://satijalab.org/seurat/reference/SeuratTheme.html" class="external-link">NoLegend</a></span><span class="op">(</span><span class="op">)</span> <span class="op">+</span> 
  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/theme.html" class="external-link">theme</a></span><span class="op">(</span>plot.title <span class="op">=</span> <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/element.html" class="external-link">element_text</a></span><span class="op">(</span>size <span class="op">=</span> <span class="fl">15</span>, face <span class="op">=</span> <span class="fl">2</span>, hjust <span class="op">=</span> <span class="fl">0.5</span><span class="op">)</span><span class="op">)</span>
<span class="va">p2</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://satijalab.org/seurat/reference/SpatialPlot.html" class="external-link">SpatialDimPlot</a></span><span class="op">(</span><span class="va">cluster4</span>, group.by <span class="op">=</span> <span class="st">"sca.snn_res.1"</span>, pt.size.factor <span class="op">=</span> <span class="fl">3</span>, 
                     label <span class="op">=</span> <span class="cn">TRUE</span>, label.size <span class="op">=</span> <span class="fl">5</span><span class="op">)</span> <span class="op">+</span> 
  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/labs.html" class="external-link">ggtitle</a></span><span class="op">(</span><span class="st">"SCA Clustering"</span><span class="op">)</span> <span class="op">+</span> <span class="fu"><a href="https://satijalab.org/seurat/reference/SeuratTheme.html" class="external-link">NoLegend</a></span><span class="op">(</span><span class="op">)</span> <span class="op">+</span> 
  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/theme.html" class="external-link">theme</a></span><span class="op">(</span>plot.title <span class="op">=</span> <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/element.html" class="external-link">element_text</a></span><span class="op">(</span>size <span class="op">=</span> <span class="fl">15</span>, face <span class="op">=</span> <span class="fl">2</span>, hjust <span class="op">=</span> <span class="fl">0.5</span><span class="op">)</span><span class="op">)</span>
<span class="va">p1</span> <span class="op">+</span> <span class="va">p2</span></code></pre></div>
<p><img src="get_started_files/figure-html/cluster-4-1.png" width="768"></p>
<p>We can examine the differentially expressed genes between cluster 10 and 14 from SCA clustering. We see that these are heterogeneous populations marked by different gene expressions.</p>
<div class="sourceCode"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">de.genes</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://satijalab.org/seurat/reference/FindMarkers.html" class="external-link">FindMarkers</a></span><span class="op">(</span><span class="va">cluster4</span>, <span class="fl">14</span>, <span class="fl">10</span>, group.by <span class="op">=</span> <span class="st">"sca.snn_res.1"</span>, verbose <span class="op">=</span> <span class="cn">FALSE</span><span class="op">)</span>
<span class="va">de.genes</span> <span class="op">&lt;-</span> <span class="va">de.genes</span><span class="op">[</span><span class="va">de.genes</span><span class="op">$</span><span class="va">p_val_adj</span> <span class="op">&lt;</span> <span class="fl">0.05</span>, <span class="op">]</span>
<span class="va">de.genes</span> <span class="op">&lt;-</span> <span class="va">de.genes</span><span class="op">[</span><span class="fu"><a href="https://rdrr.io/r/base/order.html" class="external-link">order</a></span><span class="op">(</span><span class="va">de.genes</span><span class="op">$</span><span class="va">avg_log2FC</span><span class="op">)</span>,<span class="op">]</span>
<span class="va">cluster4</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://satijalab.org/seurat/reference/ScaleData.html" class="external-link">ScaleData</a></span><span class="op">(</span><span class="va">cluster4</span>, features <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/colnames.html" class="external-link">rownames</a></span><span class="op">(</span><span class="va">de.genes</span><span class="op">)</span><span class="op">)</span>
<span class="va">p1</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://satijalab.org/seurat/reference/DoHeatmap.html" class="external-link">DoHeatmap</a></span><span class="op">(</span><span class="va">cluster4</span>, <span class="fu"><a href="https://rdrr.io/r/base/colnames.html" class="external-link">rownames</a></span><span class="op">(</span><span class="va">de.genes</span><span class="op">)</span>, group.by <span class="op">=</span> <span class="st">"sca.snn_res.1"</span><span class="op">)</span>
<span class="va">p2</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://satijalab.org/seurat/reference/SpatialPlot.html" class="external-link">SpatialFeaturePlot</a></span><span class="op">(</span><span class="va">cluster4</span>, <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/utils/head.html" class="external-link">head</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/colnames.html" class="external-link">rownames</a></span><span class="op">(</span><span class="va">de.genes</span><span class="op">)</span>,<span class="fl">2</span><span class="op">)</span>, <span class="fu"><a href="https://rdrr.io/r/utils/head.html" class="external-link">tail</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/colnames.html" class="external-link">rownames</a></span><span class="op">(</span><span class="va">de.genes</span><span class="op">)</span>,<span class="fl">2</span><span class="op">)</span><span class="op">)</span>,
                         pt.size.factor <span class="op">=</span> <span class="fl">3</span><span class="op">)</span>
<span class="va">p1</span> <span class="op">+</span> <span class="va">p2</span></code></pre></div>
<p><img src="get_started_files/figure-html/de-genes-1.png" width="1152"> &lt;&gt;  </p>
</div>
<div class="section level3">
<h3 id="session-information">Session Information<a class="anchor" aria-label="anchor" href="#session-information"></a>
</h3>
<div class="sourceCode"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="https://rdrr.io/r/base/print.html" class="external-link">print</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/utils/sessionInfo.html" class="external-link">sessionInfo</a></span><span class="op">(</span><span class="op">)</span><span class="op">)</span>
<span class="co">## R version 4.0.3 (2020-10-10)</span>
<span class="co">## Platform: x86_64-apple-darwin17.0 (64-bit)</span>
<span class="co">## Running under: macOS Big Sur 10.16</span>
<span class="co">## </span>
<span class="co">## Matrix products: default</span>
<span class="co">## BLAS:   /Library/Frameworks/R.framework/Versions/4.0/Resources/lib/libRblas.dylib</span>
<span class="co">## LAPACK: /Library/Frameworks/R.framework/Versions/4.0/Resources/lib/libRlapack.dylib</span>
<span class="co">## </span>
<span class="co">## locale:</span>
<span class="co">## [1] en_US.UTF-8/en_US.UTF-8/en_US.UTF-8/C/en_US.UTF-8/en_US.UTF-8</span>
<span class="co">## </span>
<span class="co">## attached base packages:</span>
<span class="co">## [1] stats     graphics  grDevices utils     datasets  methods   base     </span>
<span class="co">## </span>
<span class="co">## other attached packages:</span>
<span class="co">## [1] ggplot2_3.3.3      SeuratObject_4.0.0 Seurat_4.0.1       rspca_0.1.0       </span>
<span class="co">## </span>
<span class="co">## loaded via a namespace (and not attached):</span>
<span class="co">##   [1] Rtsne_0.15            colorspace_2.0-0      deldir_0.2-3          ellipsis_0.3.2        ggridges_0.5.3        rprojroot_2.0.2      </span>
<span class="co">##   [7] fs_1.5.0              rstudioapi_0.13       spatstat.data_1.7-0   farver_2.0.3          leiden_0.3.6          listenv_0.8.0        </span>
<span class="co">##  [13] ggrepel_0.9.0         RSpectra_0.16-0       codetools_0.2-18      splines_4.0.3         cachem_1.0.1          knitr_1.38           </span>
<span class="co">##  [19] polyclip_1.10-0       jsonlite_1.7.2        ica_1.0-2             cluster_2.1.0         png_0.1-7             uwot_0.1.10          </span>
<span class="co">##  [25] shiny_1.5.0           sctransform_0.3.2     spatstat.sparse_2.0-0 compiler_4.0.3        httr_1.4.2            Matrix_1.3-2         </span>
<span class="co">##  [31] fastmap_1.1.0         lazyeval_0.2.2        limma_3.46.0          cli_3.2.0             later_1.1.0.1         htmltools_0.5.2      </span>
<span class="co">##  [37] tools_4.0.3           igraph_1.2.6          gtable_0.3.0          glue_1.6.2            RANN_2.6.1            reshape2_1.4.4       </span>
<span class="co">##  [43] dplyr_1.0.2           Rcpp_1.0.7            scattermore_0.7       jquerylib_0.1.3       pkgdown_2.0.2         vctrs_0.4.1          </span>
<span class="co">##  [49] nlme_3.1-151          lmtest_0.9-38         xfun_0.30             stringr_1.4.0         globals_0.14.0        mime_0.9             </span>
<span class="co">##  [55] miniUI_0.1.1.1        lifecycle_1.0.1       irlba_2.3.3           goftest_1.2-2         future_1.21.0         MASS_7.3-53          </span>
<span class="co">##  [61] zoo_1.8-8             scales_1.1.1          spatstat.core_1.65-5  ragg_0.4.0            promises_1.1.1        spatstat.utils_2.1-0 </span>
<span class="co">##  [67] parallel_4.0.3        RColorBrewer_1.1-2    yaml_2.2.1            memoise_2.0.1         reticulate_1.18       pbapply_1.4-3        </span>
<span class="co">##  [73] gridExtra_2.3         sass_0.4.0            rpart_4.1-15          stringi_1.5.3         highr_0.8             desc_1.4.1           </span>
<span class="co">##  [79] rlang_1.0.2           pkgconfig_2.0.3       systemfonts_0.3.2     matrixStats_0.57.0    evaluate_0.15         lattice_0.20-41      </span>
<span class="co">##  [85] ROCR_1.0-11           purrr_0.3.4           tensor_1.5            labeling_0.4.2        patchwork_1.1.1       htmlwidgets_1.5.3    </span>
<span class="co">##  [91] cowplot_1.1.1         tidyselect_1.1.0      parallelly_1.23.0     RcppAnnoy_0.0.18      plyr_1.8.6            magrittr_2.0.1       </span>
<span class="co">##  [97] R6_2.5.0              generics_0.1.0        DBI_1.1.0             withr_2.4.3           mgcv_1.8-33           pillar_1.4.7         </span>
<span class="co">## [103] fitdistrplus_1.1-3    survival_3.2-7        abind_1.4-5           tibble_3.0.4          future.apply_1.7.0    crayon_1.3.4         </span>
<span class="co">## [109] KernSmooth_2.23-18    spatstat.geom_1.65-5  plotly_4.9.2.2        rmarkdown_2.13        grid_4.0.3            data.table_1.13.6    </span>
<span class="co">## [115] digest_0.6.27         xtable_1.8-4          tidyr_1.1.2           httpuv_1.5.4          textshaping_0.2.1     munsell_0.5.0        </span>
<span class="co">## [121] viridisLite_0.3.0     bslib_0.3.1</span></code></pre></div>
</div>
  </div>

  <div class="col-md-3 hidden-xs hidden-sm" id="pkgdown-sidebar">

      </div>

</div>



      <footer><div class="copyright">
  <p></p>
<p>Developed by X. Steve Niu.</p>
</div>

<div class="pkgdown">
  <p></p>
<p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.0.2.</p>
  <p class="preferably">Using <a href="https://preferably.amirmasoudabdol.name/?source=footer" class="external-link">preferably</a> template.</p>
</div>

      </footer>
</div>

  


  

  </body>
</html>
